### PR TITLE
docs: describe interaction between editorconfig and ftplugin

### PR DESCRIPTION
**Pull request by znd4**
Hi, I'm not sure if the behavior I'm documenting here is actually intentional, or is meant to be guaranteed to users, but it does seem intuitive, and does match the behavior I've seen when experimenting, so I decided to create a PR in case it's worth mentioning in the docs

### CODE REVIEW

File: runtime/doc/editorconfig.txt [hunk 1]
```diff
@@ -86,4 +86,11 @@ Example: >lua
       vim.b[bufnr].foo = val
     end
 <
+
+EditorConfig overrides filetype plugins           *ftplugin-vs-editorconfig*
```
Comment (znd4) [inline runtime/doc/editorconfig.txt:90]: I'm also not sure about this help tag. Maybe `editorconfig` should come first? Maybe `*-and-*`?

Comment (wookayin) [reply runtime/doc/editorconfig.txt:90]: editorconfig-vs-ftplugin would be better.

**Comment by wookayin** [comment]
> if the behavior I'm documenting here is actually intentional, or is meant to be guaranteed to users, but it does seem intuitive

FYI, Editorconfig runs on 'BufNewFile', 'BufRead', 'BufFilePost', and the current behavior and ordering is due to BufRead (BufReadPost) taking place *after* FileType, which makes sense because editorconfig can update per-project configs (indentation, etc.), overriding global, default indentation configs.

File: runtime/doc/editorconfig.txt [hunk 2]
```diff
@@ -86,4 +86,11 @@ Example: >lua
       vim.b[bufnr].foo = val
     end
 <
+
+EditorConfig overrides filetype plugins           *editorconfig-vs-ftplugin*
```
Comment (gpanders) [inline runtime/doc/editorconfig.txt:90]: I don't think this needs a separate section or its own tag. I would just update the sentence near the beginning of the help file.

For example, change "When a file is opened" to "When a file is opened, after running |ftplugin|s,".

Comment (znd4) [reply runtime/doc/editorconfig.txt:90]: > For example, change \"When a file is opened\" to \"When a file is opened, after running |ftplugin|s,\".

Ooh, nice. I have much to learn about technical documentation.

**Comment by gpanders** [comment]
@znd4 Is this ready for review?

**Comment by wookayin** [comment]
One note is that `after/ftplugin` would be executed after editorconfig, and it'd be even better if we can include this information as well in the documentation.

**Comment by dundargoc** [comment]
Superseded by https://github.com/neovim/neovim/pull/28736

**Pull request closed without merging**